mesonbuild/interpreter/type_checking.py

```diff
@@ -701,6 +708,11 @@ def _convert_darwin_versions(val: T.List[T.Union[str, int]]) -> T.Optional[T.Tup
         convertor=lambda x: x.lower() if isinstance(x, str) else None,
         validator=_validate_win_subsystem,
     ),
+    KwargInfo(
+        'android_exe_type',
+        (str, NoneType),
+        validator=_validate_android_exe_type,
```

Comment:
```suggestion
        validator=in_set_validator({'application', 'executable'}),
```
and drop the hand rolled validator.